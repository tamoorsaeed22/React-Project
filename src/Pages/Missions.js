import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Header from '../Components/Header';
import { fetchMissions } from '../app/features/missionsReducer';
import styles from './Missions.module.css';

const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions);
  useEffect(() => {
    dispatch(fetchMissions());
  }, [dispatch]);

  return (
    <>
      <Header />
      <table>
        <thead>
          <tr>
            <td>Name</td>
            <td>Description</td>
            <td>Status</td>
            <td />
          </tr>
        </thead>
        <tbody>
          {missions.map((missions) => (
            <tr key={missions.id}>
              <td className={styles.mission_name}>{missions.mission_name}</td>
              <td>{missions.description}</td>
              <td>NOT A MEMBER</td>
              <td><button type="button">Join Mission</button></td>
            </tr>
          ))}
        </tbody>
      </table>
    </>
  );
};

export default Missions;