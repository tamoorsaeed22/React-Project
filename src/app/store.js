import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './features/rocketsReducer';
import missionsReducer from './features/missionReducer';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducer,
  },
});

export default store;
