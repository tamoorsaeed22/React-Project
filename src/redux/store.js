import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rocket/rocketsReducer';
import missionsReducer from './mission/missionReducer';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducer,
  },
});

export default store;
