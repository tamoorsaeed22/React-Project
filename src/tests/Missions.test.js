import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../app/store';
import Missions from '../Pages/Missions';

describe('Missions Test', () => {
  it('renders correctly', () => {
    const missionsTest = renderer.create(
      <Provider store={store}>
        <Router>
          <Missions />
        </Router>
      </Provider>,
    ).toJSON();
    expect(missionsTest).toMatchSnapshot();
  });
});
