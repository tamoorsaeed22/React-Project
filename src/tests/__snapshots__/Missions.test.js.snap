// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Test renders correctly 1`] = `
Array [
  <header>
    <div
      className="header-logo-and-name"
    >
      <img
        alt="planet logo"
        className="header-logo"
        src="planet.png"
      />
      <h1>
        Space Travelers' Hub
      </h1>
    </div>
    <nav>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "fontWeight": "bold",
            "textDecoration": "underline",
          }
        }
      >
        Rockets
      </a>
      <a
        className=""
        href="/Missions"
        onClick={[Function]}
        style={
          Object {
            "fontWeight": "normal",
            "textDecoration": "none",
          }
        }
      >
        Missions
      </a>
      <a
        className="profile"
        href="/MyProfile"
        onClick={[Function]}
        style={
          Object {
            "fontWeight": "normal",
            "textDecoration": "none",
          }
        }
      >
        My Profile
      </a>
    </nav>
  </header>,
  <table>
    <thead>
      <tr>
        <td>
          Name
        </td>
        <td>
          Description
        </td>
        <td>
          Status
        </td>
        <td />
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
